src/MongoDB.Driver/AggregateFluentBase.cs

```diff
@@ -203,6 +203,12 @@ public virtual IOrderedAggregateFluent<TResult> ThenBy(SortDefinition<TResult> n
             throw new NotImplementedException();
         }
 
+        /// <inheritdoc />
+        public abstract IAggregateFluent<TNewResult> UnionWith<TForeignDocument, TNewResult>(
```

Comment:
Adding an abstract class is a breaking change.

Make it virtual and throw NotImplementedException.

Reply:
good point. will do it in the next commit

Reply:
done